Create intial page for `CS8355` (#38699)
dotnet/docs: docs/csharp/language-reference/compiler-messages/cs8355.md

---
description: "Compiler Error CS8355"
title: "Compiler Error CS8355"
ms.date: 12/10/2023
f1_keywords:
  - "CS8355"
helpviewer_keywords:
  - "CS8355"
---
# Compiler Error CS8355

An in parameter cannot have the Out attribute.

> The parameter of a method can't be set as being passed by value, while informing interop marshaler to treat this parameter as passed by reference to the native code.

Reviewer: I'd reword this a bit. `in` parameters are (generally) passed by reference:

```suggestion
The parameter of a method can't be declared as read only, while informing interop marshaler to treat this parameter as writable by the native code.
```
